Sources/SwiftKafka/RDKafka/RDKafkaConfig.swift

```diff
@@ -63,46 +65,119 @@ struct RDKafkaConfig {
         }
     }
 
-    /// A Swift wrapper for `rd_kafka_conf_set_dr_msg_cb`.
-    /// Defines a function that is called upon every message acknowledgement.
+    /// Registers passed closures as callbacks and sets the application's opaque pointer that will be passed to callbacks
+    /// - Parameter type: Kafka client type: `Consumer` or `Producer`
     /// - Parameter configPointer: An `OpaquePointer` pointing to the `rd_kafka_conf_t` object in memory.
-    /// - Parameter callback: A closure that is invoked upon message acknowledgement.
-    /// - Returns: A ``CapturedClosure`` object that must me retained by the caller as long as acknowledgements are received.
-    static func setDeliveryReportCallback(
+    /// - Parameter deliveryReport: A closure that is invoked upon message acknowledgement.
+    /// - Parameter logger: Logger instance
+    /// - Returns: A ``CapturedClosures`` object that must me retained by the caller as long as it exists.
+    static func setCallbackClosures(
+        type: RDKafka.ClientType,
```

Comment:
Why do we need the `type` here? We already have the `deliveryReport` as an optional.

Reply:
Added a guard on upper level. I just want to check that callback accidentally is not set for consumer client.